### PR TITLE
MKS CANable V2.0

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
add new board MKS CANable V2.0 Please refer to added documentation for more information.
The hardware of the board can be dound under: https://github.com/makerbase-mks/CANable-MKS

### CODE REVIEW

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen You will see an overlay for this board in CANnectivity later. ;) Thus please take a look too.

**Review by nordicjm — COMMENTED** [review]
Board is OK, please fix ordering issue

**Comment by KozhinovAlexander** [comment]
> Board is OK, please fix ordering issue

Thank you for you review. Great to see a feedback from hwv2 guy :) I'll do necessary changes today late night.

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
+.. zephyr:board:: mks_canable_v20
+
+Overview
+********
+
+The Makervase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/doc/index.rst:6]:
```suggestion
The Makerbase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+- Three LEDs: red/power_led (D1), blue/stat_led (D2), green/word_led (D3)
+- One push-button for RESET
+
+The LED red/power_led (D1) is connected directly to on-board 3.3 V and not controllable by the MCU.
+
+More information about the board can be found at the `MKS CANable V2.0 website`_.
+It is very advisable to take a look in on user manual `MKS CANable V2.0 User Manual`_ and
+schematic `MKS CANable V2.0 schematic`_ before start.
+
+- Development support: serial wire debug (SWD), JTAG, Embedded Trace Macrocell.
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/doc/index.rst:24]: This bullet point looks isolated. Should it be grouped in above list ? Or maybe rephrased ?

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+:zephyr:code-sample:`hello_world`.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: mks_canable_v20
+   :goals: build flash
+   :west-args: -S rtt-console
+   :compact:
+
+The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protoc`ol.
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/doc/index.rst:122]: Typo ?
```suggestion
The argument ``-S rtt-console`` is needed for debug purposes with SEGGER RTT protocol.
```

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+This option may is optional and may be omitted. Omitting it makes more RAM free,
+but drops RTT support.
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/doc/index.rst:124]:
```suggestion
This option is optional and may be omitted. Omitting it frees up RAM space but prevents RTT usage.
```

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+If option ``-S rtt-console`` were chosen, the connection to the target can be
+established as follows:
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/doc/index.rst:127]:
```suggestion
If option ``-S rtt-console`` is selected, the connection to the target can be established as follows:
```

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	};
+
+	aliases {
+		led0 = &green_led;
+	};
+};
+
+&clk_hse {
+	status = "disabled";
+};
```
Comment (erwango) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.dts:46]: Not required, hse (and clocks in general) is disabled by default

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts (hunk 6 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:46]: done in 89b786829c8f646f7c172890186c67f52a357ce4

**Review by nordicjm — COMMENTED** [review]
Updates need squashing i.e. 2 commits

**Comment by KozhinovAlexander** [comment]
> Updates need squashing i.e. 2 commits

Thank you. It is done now.

**Comment by nordicjm** [comment]
> > Updates need squashing i.e. 2 commits
> 
> Thank you. It is done now.

So the second vendor update commit needs squashing into the first vendor update commit and the second board commit needs squashing into the first board commit. Something like this:
```
git rebase -i 40cd35e56d43716bdbdeafafa6c356407121ca2e
  change first to edit, third to fixup, forth to drop
git cherry-pick ddf9bfdf02006c2dc4e2012ae5ad93a3a0fde584
git rebase --continue
git rebase -i 40cd35e56d43716bdbdeafafa6c356407121ca2e
  change second to fixup
```
Now you should have 2 commits

**Comment by KozhinovAlexander** [comment]
> > > Updates need squashing i.e. 2 commits
> > 
> > Thank you. It is done now.
> 
> So the second vendor update commit needs squashing into the first vendor update commit and the second board commit needs squashing into the first board commit. Something like this:
> ```
> git rebase -i 40cd35e56d43716bdbdeafafa6c356407121ca2e
>   change first to edit, third to fixup, forth to drop
> git cherry-pick ddf9bfdf02006c2dc4e2012ae5ad93a3a0fde584
> git rebase --continue
> git rebase -i 40cd35e56d43716bdbdeafafa6c356407121ca2e
>   change second to fixup
> ```
> Now you should have 2 commits

Got it. Will do.

**Review by nordicjm — DISMISSED** [review]

File: boards/makerbase/index.rst [hunk 7]
```diff
@@ -0,0 +1,10 @@
+.. _boards-makerbase:
+
+Makerbase
```
Comment (henrikbrixandersen) [inline boards/makerbase/index.rst:3]:
```suggestion
MakerBase
```

File: boards/makerbase/mks_canable_v20/board.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+board:
+  name: mks_canable_v20
+  full_name: MKS Canable V2.0
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/board.yml:3]:
```suggestion
  full_name: MKS CANable V2.0
```

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,173 @@
+.. zephyr:board:: mks_canable_v20
+
+Overview
+********
+
+The Makerbase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:6]:
```suggestion
The MakerBase MKS CANable V2.0 board features an ARM Cortex-M4 based STM32G431C8 MCU
```

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+with a wide range of connectivity support and configurations.
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:7]: A range of of? Not on this board, so I think that can be removed from the board description.

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+The default configuration can be found in the defconfig file:
+:zephyr_file:`boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig`
+
+
+Connections and IOs
+===================
+
+The stm32g431c8 mcu is the core of MKS CANable V2.0 board and it has 6 GPIO controllers.
+These controllers are responsible for pin muxing, input/output, pull-up, etc.
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:61]: This is not that important for the board configuration and could be removed.

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+For more details please refer to `MKS CANable V2.0 schematic`_.
+
+Default Zephyr Peripheral Mapping is fixed by `MKS CANable V2.0 schematic`_:
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:65]: What does this mean?

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+- SWDIO : PA13
+- SWCLK : PA14
+- NRST : PG10
+
+System Clock
+------------
+
+The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
+By default system clock is driven by PLL clock at 170 MHz,
+the PLL is driven by the 16 MHz high speed internal oscillator.
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:85]: At which frequency are you clocking the CAN controllers?

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+   The debugger is not the part of the board!
+
+Applications for the ``mks_canable_v20`` board target can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+The board is configured to be flashed using west OpenOCD runner.
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:103]: We usually don't mention this in the board documentation, as other runners can be added and used as well.

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+You should see the following message on the console:
+
+.. code-block:: console
+
+   $ Hello World! mks_canable_v20/stm32g431xx
+
+.. note::
+
+   Current OpenOCD config will skip Segger RTT for OpenOCD under 0.12.0.
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:138]: I would remove all of this and just refer to the common documentation on how to use Segger RTT.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+ * Copyright (c) 2024 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/g4/stm32g431X8.dtsi>
+#include <st/g4/stm32g431c(6-8-b)tx-pinctrl.dtsi>
+
+/ {
+	model = "Makerbase MKS CANable V2.0";
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.dts:11]:
```suggestion
	model = "MakerBase MKS CANable V2.0";
```

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+zephyr_udc0: &usb {
+	pinctrl-0 = <&usb_dm_pa11 &usb_dp_pa12>;
+	pinctrl-names = "default";
+	clocks = <&rcc STM32_CLOCK(APB1, 23U)>,
+		 <&rcc STM32_SRC_HSI48 CLK48_SEL(0)>;
+	status = "okay";
+
+	cdc_acm_uart0: cdc_acm_uart0 {
+		compatible = "zephyr,cdc-acm-uart";
+	};
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.dts:128]: This should be removed.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+		 <&rcc STM32_SRC_LSI RTC_SEL(2)>;
+};
+
+&iwdg {
+	status = "okay";
+};
+
+&die_temp {
+	status = "okay";
+};
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.dts:99]: Have you tested RNG, RTC, IWDG, and die temperature sensor? If not, please remove them.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		boot_partition: partition@0 {
+			label = "mcuboot";
+			reg = <0x00000000 DT_SIZE_K(16)>;
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.dts:109]: I doubt you will be able to fit MCUboot into 16k flash for this board? Have you tested it?

File: boards/makerbase/mks_canable_v20/mks_canable_v20.yaml [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+  - zephyr
+  - gnuarmemb
+  - xtools
+ram: 32
+flash: 64
+supported:
+  - can
+  - gpio
+  - rng
+  - counter
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.yaml:15]: Have you ran the tests for these?

File: boards/makerbase/mks_canable_v20/mks_canable_v20.yaml [hunk 21]
```diff
@@ -0,0 +1,20 @@
...
+  - pinctrl
+  - usb_device
+  - usbd
+  - watchdog
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20.yaml:19]: Same here?

File: boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig [hunk 22]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2024 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_GPIO=y
+CONFIG_ARM_MPU=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_ENTROPY_GENERATOR=y
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig:7]: Why is this enabled?

File: boards/makerbase/mks_canable_v20/support/openocd.cfg [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+if {[version_compare $current_version "0.12.0"] >= 0} {
+	# OpenOCD version is 0.12.0 or newer
+	init
+	rtt setup 0x20000000 10000 "SEGGER RTT"
+	rtt start
+	rtt server start 9090 0
+} else {
+	# OpenOCD version is older than 0.12.0
+	echo "Warn: Segger RTT is not supported for this OpenOCD version: $current_version"
+}
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/support/openocd.cfg:37]: This looks very generic and should probably not just be set for one board. It could go into a common OpenOCD configuration file.

File: dts/bindings/vendor-prefixes.txt [hunk 24]
```diff
@@ -395,6 +395,7 @@ lxa	Linux Automation GmbH
 m5stack	M5Stack
 macnica	Macnica Americas
 madmachine	Shenzhen FeiKaiTe Technology Co., Ltd.
+makerbase	Makerbase Co., Ltd.
```
Comment (henrikbrixandersen) [inline dts/bindings/vendor-prefixes.txt:398]:
```suggestion
makerbase	MakerBase Co., Ltd.
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 25]
```diff
@@ -0,0 +1,173 @@
...
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| USB       | on-chip    | universal-serial-bus                |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| PWM       | on-chip    | pwm                                 |
```
Comment (henrikbrixandersen) [inline boards/makerbase/mks_canable_v20/doc/index.rst:46]: Where is PWM used on this board?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: boards/makerbase/index.rst (hunk 7 above)
Comment (KozhinovAlexander) [reply boards/makerbase/index.rst:3]: They call them `Makerbase`, please take a look here: https://makerbase.com.cn/en/products/

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 10 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:7]: MCU supports `a range of...`. But in terms of board I agree and will update

File: boards/makerbase/mks_canable_v20/support/openocd.cfg (hunk 23 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/support/openocd.cfg:37]: Do you mean by `common` the configuration file delivered with OpenOCD itself?

File: dts/bindings/vendor-prefixes.txt (hunk 24 above)
Comment (KozhinovAlexander) [reply dts/bindings/vendor-prefixes.txt:398]: They call themselve `Makerbase`, please take a look here: https://makerbase.com.cn/en/products/

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 25 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:46]: Planned to use it for PWM LED since i see LEDs on th board as way to bright. But for the current PR it may be removed - agree.

File: boards/makerbase/mks_canable_v20/support/openocd.cfg (hunk 23 above)
Comment (henrikbrixandersen) [reply boards/makerbase/mks_canable_v20/support/openocd.cfg:37]: I meant either `boards/common/openocd.board.cmake`. I would suggest submitting this in a separate PR.

File: boards/makerbase/index.rst (hunk 7 above)
Comment (henrikbrixandersen) [reply boards/makerbase/index.rst:3]: Apparently they cannot make up their mind regarding this - I looked them up on their online store, they use "MakerBase" there: https://makerbase3d.com

Let's stick with "Makerbase".

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 12 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:65]: Default Zephyr Peripheral Mapping.

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 13 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:85]: This part describes main MCU clock and not a CAN clock. 
I will add a CAN clock description.

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 14 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:103]: Changed to: The board could be flashed using west.

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 15 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:138]: Somehow I can't find proper compact description for segger RTT usage same way it is done in my current documentation. Would you please put a link here?

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts (hunk 17 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:128]: done

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts (hunk 18 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:99]: done

File: boards/makerbase/mks_canable_v20/mks_canable_v20.yaml (hunk 20 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.yaml:15]: Please correct me: But CI shall at least build them.

File: boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig (hunk 22 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20_defconfig:7]: Good catch - copy-paste from other board.

File: boards/makerbase/mks_canable_v20/support/openocd.cfg (hunk 23 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/support/openocd.cfg:37]: > I meant `boards/common/openocd.board.cmake`. I would suggest submitting this in a separate PR.

It is a good advice. Since it wil blow up the current scope and I can not test it for other boards  - I would prefer to leave it here first. After merging this PR I would reallocate it to boards/common/openocd.board.cmake file. Or you can fell free to do so. I can support.

File: boards/makerbase/index.rst [hunk 26]
```diff
@@ -0,0 +1,10 @@
...
+##################
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/index.rst:4]: Thank you

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts (hunk 19 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:89]: Will use only flash0

File: boards/makerbase/mks_canable_v20/mks_canable_v20.yaml (hunk 21 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.yaml]: will be cleaned

**Comment by KozhinovAlexander** [comment]
> Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?

There are a lot of differences betwee each version of the HW. To me their versioning seems more as a tagging of evolution and not of progress.

**Comment by kartben** [comment]
> > Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?
> 
> There are a lot of differences between each version of the HW. To me their versioning seems more as an tagging of evolution approach and not of progress.

Yes, this. I very much agree with the point of view that we shouldn't try to "force" boards into hwmv2 variants/revisions just because naming scheme from a given vendor *seems* to call for using them. Sometimes ProductFoo and ProductFoo X just happen to be very different beasts (see ex how we have `rpi_pico2` coming up as its own board).

**Comment by nordicjm** [comment]
> > > Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?
> > 
> > 
> > There are a lot of differences between each version of the HW. To me their versioning seems more as an tagging of evolution approach and not of progress.
> 
> Yes, this. I very much agree with the point of view that we shouldn't try to "force" boards into hwmv2 variants/revisions just because naming scheme from a given vendor _seems_ to call for using them. Sometimes ProductFoo and ProductFoo X just happen to be very different beasts (see ex how we have `rpi_pico2` coming up as its own board).

How does it not fit? The schematics for v1.00 and v1.00 pro are the same except one has a TJA1051T and the other has a ADM3053 on page 2 of the schematic, the other 5 pages of the schematics are exactly the same. That's the definition of a board revision. The v2.00 seems to be completely different but that doesn't mean v1.00 and v1.00 pro can't use board revisions

**Comment by kartben** [comment]
> > > > Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?
> > > 
> > > 
> > > There are a lot of differences between each version of the HW. To me their versioning seems more as an tagging of evolution approach and not of progress.
> > 
> > Yes, this. I very much agree with the point of view that we shouldn't try to "force" boards into hwmv2 variants/revisions just because naming scheme from a given vendor _seems_ to call for using them. Sometimes ProductFoo and ProductFoo X just happen to be very different beasts (see ex how we have `rpi_pico2` coming up as its own board).
> 
> How does it not fit? The schematics for v1.00 and v1.00 pro are the same except one has a TJA1051T and the other has a ADM3053 on page 2 of the schematic, the other 5 pages of the schematics are exactly the same. That's the definition of a board revision. The v2.00 seems to be completely different but that doesn't mean v1.00 and v1.00 pro can't use board revisions. And rpi_pico2 is a different board, no-where as anyone or the company themselves said it's just a revised PCB 

But this is the PR introducing **V2**, so I was commenting on that, Jamie. And you seem to agree it's completely different, and pretty the same story as Pico vs. Pico 2?

**Comment by nordicjm** [comment]
> > > > > Given that upstream threats this as one of multiple board variants (see https://github.com/makerbase-mks/CANable-MKS), I think we might want to do the same (bu renaming the board folder etc.). Support for the V1.0 and the PRO editions can then be added later on. What do you think?
> > > > 
> > > > 
> > > > There are a lot of differences between each version of the HW. To me their versioning seems more as an tagging of evolution approach and not of progress.
> > > 
> > > 
> > > Yes, this. I very much agree with the point of view that we shouldn't try to "force" boards into hwmv2 variants/revisions just because naming scheme from a given vendor _seems_ to call for using them. Sometimes ProductFoo and ProductFoo X just happen to be very different beasts (see ex how we have `rpi_pico2` coming up as its own board).
> > 
> > 
> > How does it not fit? The schematics for v1.00 and v1.00 pro are the same except one has a TJA1051T and the other has a ADM3053 on page 2 of the schematic, the other 5 pages of the schematics are exactly the same. That's the definition of a board revision. The v2.00 seems to be completely different but that doesn't mean v1.00 and v1.00 pro can't use board revisions. And rpi_pico2 is a different board, no-where as anyone or the company themselves said it's just a revised PCB
> 
> But this is the PR introducing **V2**, so I was commenting on that, Jamie. And you seem to agree it's completely different, and pretty the same story as Pico vs. Pico 2?

For this board yes, the original comment refers to v1.00 and pro

**Review by nordicjm — DISMISSED** [review]

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: boards/makerbase/mks_canable_v20/doc/index.rst [hunk 27]
```diff
@@ -0,0 +1,171 @@
...
+
+System Clock
+------------
+
+The MKS CANable V2.0 system clock is driven by internal high speed oscillator.
+By default system clock is driven by PLL clock at 170 MHz,
+the PLL is driven by the 16 MHz high speed internal oscillator.
+
+The FDCAN1 peripheral is driven by PLLQ, which has 170 MHz frequency sam as main
+system clock.
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:83]: > The CAN FD peripheral should not run at 170 MHz. Please aim for a CAN core frequency of 80 MHz for interoperability reasons (recommended by CiA).
> 

This is a good point. I see it in CiA 601-2 CAN  and here: [CiA 601-2 CAN ](https://www.can-cia.org/fileadmin/cia/documents/publications/cnlm/march_2018/18-1_p28_recommendation_for_the_canfd_bit-timing_holger_zeltwanger_cia.pdf)

Unfortunatel I can not find other way than setting main clock to 160MHz s.t. CAN clock will be 80MHz. But since the board is specifically aimed for CAN it would be totally fine.

> Also, what does "sam" mean in the sentence above?

Shall be "same". Corrected.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+			gpios = <&gpioa 15 GPIO_ACTIVE_LOW>;
+			label = "blue-status D2";
+		};
+		green_led: led_3 {
+			gpios = <&gpioa 0 GPIO_ACTIVE_LOW>;
+			label = "green-word D3";
+		};
+	};
+
+	aliases {
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:38]: Will be removed. Thanks.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+		led0 = &green_led;
+	};
+};
+
+&clk_lsi {
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:43]: No. I will remove it. I think I shall alsoe remove other unused entities. I will do.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+	status = "okay";
+};
+
+&timers2 {
+	/* Timer with 1us tick used for CAN timestamp */
+	st,prescaler = <169>;
+	counters2: counter {
+		status = "okay";
+	};
+};
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:86]: I see. I will move it to an overlay s.t. later cannectivity will get this overlay.

File: boards/makerbase/mks_canable_v20/mks_canable_v20.dts [hunk 31]
```diff
@@ -0,0 +1,102 @@
...
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,canbus = &fdcan1;
+	};
+
+	aliases {
+		led0 = &blue_led;
+		led1 = &green_led;
+		mcuboot-led0 = &blue_led;
```
[the remark this answers is not in the document]
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/mks_canable_v20.dts:23]: Removed

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen Interestingly I can not answer to this your message:
> I still do not think this belongs in a board specific configuration. By introducing this now and "fixing it later", we risk this will be copied to multiple boards, making the cleanup task much bigger.

If you are not up for the task of adding this as a generic functionality now, I suggest leaving it out.

I see, what I can do. Need to test it first for the other boards.

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen Thank you for your input. I've done the chnages requested. The OpenOCD config will be moved to separate PR in few minutes.

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 15 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:135]: This one shows only KConfig entry and no step-by-step guide like i my documentation.

File: boards/makerbase/mks_canable_v20/support/openocd.cfg (hunk 23 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/support/openocd.cfg]: I've dropped it and will prepare separate PR.

File: boards/makerbase/mks_canable_v20/doc/index.rst (hunk 15 above)
Comment (henrikbrixandersen) [reply boards/makerbase/mks_canable_v20/doc/index.rst:135]: Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/doc/index.rst:135]: > Feel free to open a PR for improving the documentation on using Segger RTT console, but let's make it generic instead of board-specific (as there is really nothing board-specific to it).

Definitely I could take over with my current experience. Would you please suggest a documentation section (please a link to some index.rst file), where I can place the corresponding documentation?

File: boards/makerbase/mks_canable_v20/support/openocd.cfg (hunk 23 above)
Comment (KozhinovAlexander) [reply boards/makerbase/mks_canable_v20/support/openocd.cfg]: Segger RTT section is moved to more general approach within the PR #82413

**Review by henrikbrixandersen — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by kartben** [comment]
@erwango please revisit

**Review by erwango — APPROVED** [review]
